tests/end2end-tests/data_process/gen_test_graphs.py

```diff
@@ -0,0 +1,374 @@
+    if args.graph_size == 'tiny':
+        factor = factor * 10**0
+    elif args.graph_size == 'small':
+        factor = factor * 10**2
+    elif args.graph_size == 'medium':
+        factor = factor * 10**4
+    elif args.graph_size == 'big':
+        factor = factor * 10**5
+    elif args.graph_size == 'large':
+        factor = factor * 10**6
+        
+    n_dfs_dict = {}
+    # -------- build node1 table ---------
+    # create node id strings 
+    n1ids = np.random.choice(np.arange(NUM_NTYPE1 * factor * 2), NUM_NTYPE1 * factor, replace=False)
+    n1_ids = ['n1id-' + str(i) for i in n1ids]
+    # n1_ids = ids
+    # create two feature columns
+    f1 = np.random.randn(NUM_NTYPE1 * factor)
+    f2 = np.random.randint(0, 10, size=NUM_NTYPE1 * factor)
+    # create the node 1 pandas dataframe
+    n1_df = pd.DataFrame({'node1-id': n1_ids,
+                          'f1': f1,
+                          'f2': f2})
+    n_dfs_dict['n0'] = n1_df
+
+    # -------- build node2 table ---------
+    # create node id strings
+    n2ids = np.random.choice(np.arange(NUM_NTYPE2 * factor * 2), NUM_NTYPE2 * factor, replace=False)
+    n2_ids = ['n2id-' + str(i) for i in n2ids]
+    # n2_ids = ids
+    # create three feaure columns
+    f1 = np.random.randn(NUM_NTYPE2 * factor)
+    f2 = np.random.randint(0, 2, size=NUM_NTYPE2 * factor)
+    f3 = np.random.rand(NUM_NTYPE2 * factor)
+    # create the node 2 pandas dataframe
+    n2_df = pd.DataFrame({'node2-id': n2_ids,
+                          'f1': f1,
+                          'f2-label': f2,       # labels for all nodes
+                          'f3': f3})
+    n_dfs_dict['n1'] = n2_df
+
+    # -------- build node3 table ---------
+    # create node id strings
+    n3ids = np.random.choice(np.arange(NUM_NTYPE3 * factor * 2), NUM_NTYPE3 * factor, replace=False)
+    n3_ids = ['n3id-' + str(i) for i in n3ids]
+    # n3_ids = ids
+    # create the node 3 pandas dataframe
+    n3_df = pd.DataFrame({'node3-id': n3_ids})
+    n_dfs_dict['n2'] = n3_df
+    
+    e_dfs_dict = {}
+    # -------- build edge1 between node1 and node2 ---------
+    e1_src_ids = np.random.choice(n1_ids, NUM_ETYPE1 * factor + 100, replace=True)
+    e1_dst_ids = np.random.choice(n2_ids, NUM_ETYPE1 * factor + 100, replace=True)
+    e1_df = pd.DataFrame({'src_id': e1_src_ids,
+                          'dst_id': e1_dst_ids})
+    e1_df = e1_df.drop_duplicates().iloc[ :NUM_ETYPE1 * factor]
+    e_dfs_dict[('n0', 'e0', 'n1')] = e1_df
+
+    # -------- build edge2 between node1 and node3 ---------
+    e2_src_ids = np.random.choice(n1_ids, NUM_ETYPE2 * factor + 100, replace=True)
+    e2_dst_ids = np.random.choice(n3_ids, NUM_ETYPE2 * factor + 100, replace=True)
+    e2_df = pd.DataFrame({'src_id': e2_src_ids,
+                          'dst_id': e2_dst_ids})
+    e2_df = e2_df.drop_duplicates().iloc[ :NUM_ETYPE2 * factor]
+    
+    # [TODO] James: Keep these commented codes for later test after construct graph fix this issue.
```

Comment:
what is the issue?

Reply:
This is the issue about only partial nodes have labels.